src/Bridges/Constraint/norm_spec_nuc_to_psd.jl

```diff
@@ -0,0 +1,167 @@
+function concrete_bridge_type(::Type{<:NormSpectralBridge{T}}, G::Type{<:MOI.AbstractVectorFunction}, ::Type{MOI.NormSpectralCone}) where T
+    F = MOIU.promote_operation(vcat, T, MOIU.scalar_type(G), T)
+    return NormSpectralBridge{T, F, G}
+end
+
+# Attributes, Bridge acting as a model
+MOI.get(b::NormSpectralBridge{T, F, G}, ::MOI.NumberOfConstraints{F, MOI.PositiveSemidefiniteConeTriangle}) where {T, F, G} = 1
+MOI.get(b::NormSpectralBridge{T, F, G}, ::MOI.ListOfConstraintIndices{F, MOI.PositiveSemidefiniteConeTriangle}) where {T, F, G} = [b.psd_index]
+
+# References
+MOI.delete(model::MOI.ModelLike, c::NormSpectralBridge) = MOI.delete(model, c.psd_index)
+
+# Attributes, Bridge acting as a constraint
+function MOI.get(model::MOI.ModelLike, ::MOI.ConstraintFunction, c::NormSpectralBridge{T, F, G}) where {T, F, G}
+    psd_func = MOIU.eachscalar(MOI.get(model, MOI.ConstraintFunction(), c.psd_index))
+    t = psd_func[1]
+    side_dim = c.row_dim + c.column_dim
+    X = psd_func[[trimap(i, j) for j in 1:c.column_dim for i in (c.column_dim + 1):side_dim]]
+    return MOIU.convert_approx(G, MOIU.operate(vcat, T, t, X))
+end
+MOI.get(model::MOI.ModelLike, ::MOI.ConstraintSet, c::NormSpectralBridge) = MOI.NormSpectralCone(c.row_dim, c.column_dim)
+function MOI.get(model::MOI.ModelLike, ::MOI.ConstraintPrimal, c::NormSpectralBridge)
+    primal = MOI.get(model, MOI.ConstraintPrimal(), c.psd_index)
+    t = primal[1]
+    side_dim = c.row_dim + c.column_dim
+    X = primal[[trimap(i, j) for j in 1:c.column_dim for i in (c.column_dim + 1):side_dim]]
+    return vcat(t, X)
+end
+# Given [U X; X' V] is dual on PSD constraint, the dual on NormSpectralCone
+# constraint is (tr(U) + tr(V), 2X) in NormNuclearCone.
+function MOI.get(model::MOI.ModelLike, ::MOI.ConstraintDual, c::NormSpectralBridge)
+    dual = MOI.get(model, MOI.ConstraintDual(), c.psd_index)
+    side_dim = c.row_dim + c.column_dim
+    t = sum(dual[trimap(i, i)] for i in 1:side_dim)
+    X = 2 * dual[[trimap(i, j) for j in 1:c.column_dim for i in (c.column_dim + 1):side_dim]]
+    return vcat(t, X)
+end
+
+"""
+    NormNuclearBridge{T}
+
+The `NormNuclearCone` is representable with an SDP constraint and extra variables,
+since ``t \\ge \\sum_i \\sigma_i (X)`` if and only if there exists symmetric
+matrices ``U, V`` such that ``[U X; X' V] \\succ 0`` and ``t \\ge (tr(U) + tr(V)) / 2``.
+"""
+struct NormNuclearBridge{T, F, G, H} <: AbstractBridge
+    row_dim::Int # row dimension of X
+    column_dim::Int # column dimension of X
+    U::Vector{MOI.VariableIndex}
+    V::Vector{MOI.VariableIndex}
+    ge_index::CI{F, MOI.GreaterThan{T}}
+    psd_index::CI{G, MOI.PositiveSemidefiniteConeTriangle}
+end
+function bridge_constraint(::Type{NormNuclearBridge{T, F, G, H}}, model::MOI.ModelLike, f::H, s::MOI.NormNuclearCone) where {T, F, G, H}
+    f_scalars = MOIU.eachscalar(f)
+    row_dim = s.row_dim
+    column_dim = s.column_dim
+    @assert row_dim <= column_dim
```

Comment:
Replace by `row_dim <= column_dim || error("Reformulation for when the number of column is larger than the number of rows is not implemented yet.")`

Reply:
The current definition of the cone in sets.jl also includes this restriction. If we are to keep that restriction then we should error when the set is created rather than in the Bridge. I was proposing to keep the restriction in the cone definition.

Reply:
What is going to fail in the bridges if `row_dim <= column_dim` is not satisfied ?

Reply:
It looks like they still work actually. OK so I removed the restriction.